### DISCUSSION TITLE
Set aplication theme

### DISCUSSION THREAD

**Discussion by LuckyTurtleDev** [hecrj/iced#1257] [Q&A] [answered]
If it possible to set an application wide theme/style or do I really need to add `.style()` to every object?

Update:
I notify that Container can apply the style to some widgets as example Text but it can not apply the style to some others as example Buttons.
Why is this so?

**Comment by emann** [comment, marked as answer]
Being able to set an application-wide theme at runtime is not currently supported however is a crucial precursor to https://github.com/iced-rs/iced/issues/1022. Then all #1022 has to do on top of it is provide default "native looking" themes for each OS/platform that are the default unless a user specifies another theme. Even better if you can choose to override specific parts of the default themes. I'm very interested in implementing this so that I can use it in a project of mine.

 @hecrj, can you point me in the right direction for where to start?

**Reply by emann** [reply]
@Lukas1818 FYI I have [drafted a PR for this](https://github.com/iced-rs/iced/pull/1277)
